### PR TITLE
[Doc] Upgrade to `Docfx 2.67.0`

### LINKED ISSUE #6666: [Build System]: Frequent DocFx timeouts
**Version Information**
Version of Akka.NET? latest `dev` branch

**Describe the bug**

Our DocFx build stage repeatedly times out and has frequently done so over the past several months:

```
[23-04-24 06:24:39.234]Warning:[MetadataCommand.ExtractMetadata]Timeout (1000ms) exceeded. Killing process git.
	1 Warning(s)
	0 Error(s)
```

It's this stage of our build pipeline:

https://github.com/akkadotnet/akka.net/blob/fd4860bac99fc231b6cf2c18c761b7079af9b80b/build.fsx#L477-L506

The `treatWarningsAsErrors` flag is what causes this to fail the build pipeline - we rely on that flag in order to make sure breakage doesn't happen inside the documentation website itself (i.e. broken links.)

**Expected behavior**

Unless there's a markdown / linking / code reference error, the website build successfully each and every time.

**Actual behavior**

Frequent, but inconsistent `git` timeouts during the DocFx build stage.

**Additional context**

This could be a function of how big the repository has become and we just need to expand the DocFx timeout for this stage - to something higher than 1s.

### PR DESCRIPTION
**Pull request by eaba**
Fixes #6666

## Changes

- remove `docfx.console`

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).

### CODE REVIEW

File: build.fsx [hunk 1]
```diff
@@ -491,10 +491,12 @@ Target "DocFx" (fun _ ->
...
 
     let docsPath = FullName "./docs"
-    let docFxPath = FullName(findToolInSubPath "docfx.exe" "tools/docfx.console/tools")
+    let docFxPath = FullName(findToolInSubPath "docfx.exe" "tools")
     
     let args = StringBuilder()
                 |> append (docsPath @@ "docfx.json" )
+                //|> append ("--Docfx_Git_Timeout")
+                //|> append "30000"
                 |> append ("--warningsAsErrors")
```
Comment (eaba) [inline build.fsx]: Unknown option 'Docfx_Git_Timeout'

File: build.ps1 [hunk 2]
```diff
@@ -33,7 +33,7 @@ $FakeVersion = "4.63.0"
 $NugetVersion = "5.8.0";
 $NugetUrl = "https://dist.nuget.org/win-x86-commandline/v$NugetVersion/nuget.exe"
 $ProtobufVersion = "3.21.5"
-$DocfxVersion = "2.59.4"
+$DocfxVersion = "2.67.0"
```
Comment (eaba) [inline build.ps1:37]: Docfx 2.67.0

File: build.ps1 [hunk 3]
```diff
@@ -89,13 +89,11 @@ if (!(Test-Path $ProtobufExePath)) {
...
+$DocfxExePath = Join-Path $ToolPath "docfx.exe"
 if (!(Test-Path $DocfxExePath)) {
     Write-Host "Installing Docfx..."
-    Invoke-Expression "&`"$NugetPath`" install docfx.console -ExcludeVersion -Version $DocfxVersion -OutputDirectory `"$ToolPath`"" | Out-Null;
-    if ($LASTEXITCODE -ne 0) {
-        Throw "An error occured while restoring docfx.console from NuGet."
-    }
+	##Invoke-Expression "&dotnet new tool-manifest --force"
+    dotnet tool install docfx --version $DocfxVersion --tool-path "$ToolPath" --output "$ToolPath"
 }
```
Comment (eaba) [inline build.ps1:100]: dotnet tool install

File: docs/docfx.json [hunk 4]
```diff
@@ -77,6 +77,7 @@
             "template"
         ],
         "postProcessors": ["ExtractSearchIndex"],
-        "noLangKeyword": false
+        "noLangKeyword": false,
+        "Docfx_Git_Timeout": 30000
```
Comment (eaba) [inline docs/docfx.json:82]: I did `Docfx_Git_Timeout` for build

**Review by eaba — COMMENTED** [review]
Removed `docfx.console` @Aaronontheweb

File: docs/docfx.json (hunk 4 above)
Comment (eaba) [reply docs/docfx.json:82]: Is this right? @Aaronontheweb

**Comment by eaba** [comment]
Local.....
image

**Comment by eaba** [comment]
https://github.com/dotnet/docfx/issues/8676

File: docs/docfx.json (hunk 4 above)
Comment (Aaronontheweb) [reply docs/docfx.json:82]: You tell me :p

**Comment by Aaronontheweb** [comment]
> [dotnet/docfx#8676](https://github.com/dotnet/docfx/issues/8676)

I'd work around this issue and do a local install like how we do for Incrementalist:

https://github.com/akkadotnet/akka.net/blob/c402cc7225a334614d75c5d9039afb01e446c79f/build.ps1#L105-L113

We make zero attempts at updating the existing tool - users have to delete their `tools` folder in the root of the Akka.NET repo and re-run the install script to upgrade. I'd do that.

File: build.ps1 [hunk 5]
```diff
@@ -89,13 +89,11 @@ if (!(Test-Path $ProtobufExePath)) {
 ###########################################################################
 
 # Make sure Docfx has been installed.
-$DocfxExePath = Join-Path $ToolPath "docfx.console/tools/docfx.exe"
+$DocfxExePath = Join-Path $ToolPath "docfx.exe"
```
Comment (Aaronontheweb) [inline build.ps1]: need a sub-folder for this

File: build.fsx [hunk 6]
```diff
@@ -488,14 +488,16 @@ Target "DocFx" (fun _ ->
...
+                OutputDirectory = currentDirectory @@ "tools" }) "msdn.4.5.2" 
 
     let docsPath = FullName "./docs"
-    let docFxPath = FullName(findToolInSubPath "docfx.exe" "tools/docfx.console/tools")
+    let docFxPath = FullName(findToolInSubPath "docfx.exe" "tools/docfx/")
     
     let args = StringBuilder()
                 |> append (docsPath @@ "docfx.json" )
-                |> append ("--warningsAsErrors")
+                //|> append ("--Docfx_Git_Timeout")
```
Comment (Aaronontheweb) [inline build.fsx]: Aren't these the settings we're trying to turn on? And don't we want to keep `--warningsAsErrors`?

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]

**Review by Aaronontheweb — COMMENTED** [review]
Looks like you disabled the DocFx settings we wanted to turn on in this PR and `warningsAsErrors`

**Review by Aaronontheweb — COMMENTED** [review]
Looks like you disabled the DocFx settings we wanted to turn on in this PR and `warningsAsErrors`

File: build.fsx (hunk 6 above)
Comment (eaba) [reply build.fsx]: If I turn it back, is not working. Let me try another way @Aaronontheweb

File: docs/docfx.json (hunk 4 above)
Comment (Aaronontheweb) [reply docs/docfx.json:82]: Ah ok, so this is set inside the `docfx.json` now instead of FAKE. Makes sense.

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**